app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -3499,8 +3515,19 @@ class BrowserTabViewModel @Inject constructor(
         }
     }
 
-    fun saveReplyProxyForBlobDownload(originUrl: String, replyProxy: JavaScriptReplyProxy) {
-        replyProxyMap[originUrl] = replyProxy
+    fun saveReplyProxyForBlobDownload(originUrl: String, replyProxy: JavaScriptReplyProxy, locationHref: String? = null) {
+        appCoroutineScope.launch { // FF check has disk IO
```

Comment:
same nit: could combine launch+dispatcher selection

Reply:
I just wanted the FF to be in IO, the rest I wanted on main as we have it before